staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -170,19 +171,15 @@ func decorateResponseWriter(ctx context.Context, responseWriter http.ResponseWri
 		omitStages:     omitStages,
 	}
 
-	// check if the ResponseWriter we're wrapping is the fancy one we need
-	// or if the basic is sufficient
-	//lint:file-ignore SA1019 Keep supporting deprecated http.CloseNotifier
-	_, cn := responseWriter.(http.CloseNotifier)
-	_, fl := responseWriter.(http.Flusher)
-	_, hj := responseWriter.(http.Hijacker)
-	if cn && fl && hj {
-		return &fancyResponseWriterDelegator{delegate}
-	}
-	return delegate
+	return responsewriter.Wrap(delegate, responsewriter.Callback{
+		Hijack: func() (net.Conn, *bufio.ReadWriter, error) {
+			return delegate.hijack()
```

Comment:
The go compiler I use does not inline this call.  We could just put the body right here, nothing else calls it.

Reply:
no more callback.